browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -50,12 +55,91 @@ bool ApplyPotentialReferrerBlock(std::shared_ptr<BraveRequestInfo> ctx) {
   return false;
 }
 
+bool ApplyPotentialQueryStringFilter(std::shared_ptr<BraveRequestInfo> ctx) {
+  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  if (!ctx->request_url.has_query()) {
+    return false;
+  }
+
+  std::string new_query = ctx->request_url.query();
+
+  bool modified = false;
+  for (auto tracker : query_string_trackers) {
+    size_t key_size, key_start;
+
+    // Look for the tracker anywhere in the query string
+    // (e.g. ?...fbclid=1234...).
+    {
+      const std::string key = tracker + "=";
+      key_size = key.size();
+      key_start = new_query.find(key, 0);
+    }
+    if (key_start == std::string::npos) {
+      continue;
+    }
+
+    // Check to see if the tracker is preceded by a '&'
+    // (e.g. ?...&fbclid=1234...).
+    if (key_start != 0) {
+      if (new_query[--key_start] != '&') {
+        // Tracker is substring of another key (e.g. ?...&abc-fbclid=1234...).
+        continue;
+      }
+      key_size++;
+    }
+
+    {
+      // Find the size of the tracking parameter value.
+      const size_t value_start = key_start + key_size;
+      size_t value_end = new_query.find("&", key_start + 1);
+      if (value_end == std::string::npos) {
+        value_end = new_query.size();  // Tracker is the last query parameter.
+      }
+      if (value_end <= value_start) {
+        DCHECK_EQ(value_end, value_start);
+        continue;  // Empty value, no need to remove tracker.
+      }
+
+      // Remove tracking parameter and its value from the string.
+      modified = true;
+      new_query.erase(key_start, value_end - key_start);
+    }
+
+    // Remove the trailing '&' if we removed the first parameter
+    // (e.g. ?fbclid=1234&...).
+    if (!new_query.empty() && key_start == 0) {
+      DCHECK_EQ(new_query[0], '&');
+      new_query.erase(key_start, 1);
+    }
+
+    if (new_query.empty()) {
+      break;  // Nothing left to filter.
+    }
+  }
+
+  if (modified) {
+    url::Replacements<char> replacements;
+    if (new_query.empty()) {
+      replacements.ClearQuery();
+    } else {
+      replacements.SetQuery(new_query.c_str(),
+                            url::Component(0, new_query.size()));
+    }
+    ctx->new_url_spec =
+      ctx->request_url.ReplaceComponents(replacements).spec();
```

Comment:
wrong whitespace, I'd suggest to `clang-format` everything

Reply:
Done. There is a lot more to fix in this file, but for clarity, I'll limit my changes to the code I'm adding.